src/System.Text.CaseFolding/System/Text/CaseFolding/SimpleCaseFolding.cs

```diff
@@ -0,0 +1,606 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System.Text.CaseFolding
+{
+    /// <summary>
+    /// </summary>
+    public static partial class SimpleCaseFolding
+
+    {
+        private static ref ushort s_MapLevel1 => ref MapLevel1[0];
+        private static ref char s_refMapData => ref MapData[0];
+        private static ref ushort s_refMapSurrogateLevel1 => ref MapSurrogateLevel1[0];
+        private static ref (char highSurrogate, char lowSurrogate) s_refMapSurrogateData => ref MapSurrogateData[0];
+
+        /// <summary>
+        /// </summary>
+        public static char SimpleCaseFold(char c)
+        {
+            if (c <= 0x5ff)
+            {
+                return MapBelow5FF[c];
+            }
+
+            // var v = L1[c >> 8];
+            // var ch = L3[v + (c & 0xFF)];
+            // Still slow due to border checks.
+            var v = Unsafe.Add(ref s_MapLevel1, c >> 8);
+            var ch = Unsafe.Add(ref s_refMapData, v + (c & 0xFF));
+
+            return ch == 0 ? c : ch;
+        }
+
+        // Mapping for chars > 0x5ff slowly due to 2-level mapping.
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static int SimpleCaseFoldCompareAbove05ff(char c1, char c2, ref ushort refMapLevel1, ref char refMapData)
```

Comment:
>SimpleCaseFoldCompareAbove05ff [](start = 27, length = 30)

maybe a good idea to add asserts here for characters has to be greater than 5ff

Reply:
Added.